File: apps/web/src/lib/email.ts
import { Resend } from 'resend'

const resend = process.env.RESEND_API_KEY
  ? new Resend(process.env.RESEND_API_KEY)
  : null

const from = process.env.EMAIL_FROM ?? 'StoryTunes <onboarding@resend.dev>'

type SendEmailInput = {
  to: string
  subject: string
  text: string
  html?: string
}

/**
 * Thin transactional-email wrapper around Resend.
 *
 * In dev, if RESEND_API_KEY is unset we log to the console instead of throwing,
 * so sign-in/OTP flows are testable without credentials.
 */
export async function sendEmail({ to, subject, text, html }: SendEmailInput) {
  if (!resend) {
    console.info(`[email:dev] to=${to} subject="${subject}"\n${text}`)
    return
  }

  const { error } = await resend.emails.send({
    from,
    to,
    subject,
    text,
    ...(html ? { html } : {}),
  })

  if (error) {
    throw new Error(`Resend failed: ${error.message}`)
  }
}

File: apps/api/app/services/email.py
"""Transactional email via Resend (PDD §25). Dev-safe when key is unset."""

import resend

from app.core.config import settings

resend.api_key = settings.resend_api_key or None


def send_email(to: str, subject: str, html: str) -> str | None:
    if not settings.resend_api_key:
        print(f"[email:dev] to={to} subject={subject!r}")
        return None
    res = resend.Emails.send(
        {"from": settings.email_from, "to": to, "subject": subject, "html": html}
    )
    return res.get("id") if isinstance(res, dict) else None
